Model/AdyenRequestMerchantSession.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2017 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Model;
+
+use Adyen\Payment\Api\AdyenRequestMerchantSessionInterface;
+
+class AdyenRequestMerchantSession implements AdyenRequestMerchantSessionInterface
+{
+    /**
+     * @var \Adyen\Payment\Helper\Data
+     */
+    protected $_adyenHelper;
+
+    /**
+     * @var \Magento\Store\Model\StoreManagerInterface
+     */
+    protected $_storeManager;
+
+    /**
+     * AdyenRequestMerchantSession constructor.
+     * @param \Adyen\Payment\Helper\Data $adyenHelper
+     */
+    public function __construct(
+        \Adyen\Payment\Helper\Data $adyenHelper,
+        \Magento\Store\Model\StoreManagerInterface $storeManager
+    ) {
+        $this->_adyenHelper = $adyenHelper;
+        $this->_storeManager = $storeManager;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getMerchantSession()
+    {
+        // Works for test and live. Maybe we need to switch for validationUrl from callback event waiting for apple to respond
+        $validationUrl = "https://apple-pay-gateway-cert.apple.com/paymentservices/startSession";
+
+        // create a new cURL resource
+        $ch = curl_init();
+
+        $merchantIdentifier = $this->_adyenHelper->getAdyenApplePayMerchantIdentifier();
+
+        $domainName = $_SERVER['SERVER_NAME'];
```

Comment:
$storeParsedUrl = parse_url($this->_storeManager->getStore()->getBaseUrl(\Magento\Framework\UrlInterface::URL_TYPE_LINK))['host'];

Reply:
indeed better, lets use \Magento\Framework\UrlInterface::URL_TYPE_WEB instead of URL_TYPE_LINK so if you have rewrite off index.php is not in the domainname